src/components/controls/keyboard-controls.js

```diff
@@ -0,0 +1,94 @@
+var registerControls = require('../../core/controls').registerControls;
```

Comment:
This is vaguely my feeling on Pointer Lock as well, but I feel a little stronger about it in the keyboard case – I'd vote to aim for whatever we think the VR-equivalent of progressive enhancement is going to be. Lots of users won't have HMDs or even Cardboard around all the time, they're just clicking a fun-looking link they found on Twitter. If we remove keyboard support and require drag-mouse-to-look-around, a lot of otherwise interesting VR scenes are going to be pretty unusable a desktop PC. Lot of cases where WebVR experiences should support non-VR users if they can – something casual like Convrge, where I might want to leave the window open while I'm chatting with friends in the scene, but checking my email at the same time. Obviously a separate `keyboard-controls` repo would let devs work around that easily, but defaults are important.

On the flip-side of "defaults are important", I agree that we should strongly encourage better locomotion options. I just feel bad nuking the desktop experience to do that. Any way to keep a decent fallback experience while still discouraging WASD as the primary locomotion? If there's a middle ground then great, if not then maybe you're right and we need Tough Choices.